socket.go

```diff
@@ -51,7 +51,7 @@ func (s *SocketServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	client := newClient(getRemoteAddr(r), s.auth, session, s.broker, conn, s.editable, s.baseURL)
+	client := newClient(getRemoteAddr(r), s.auth, session, s.broker, conn, s.editable, s.baseURL, &r.Header)
```

Comment:
I'm hazy about how this works. How are the HTTP headers from the original GET (index.html) request preserved on this request? Does the browser do that automatically?

Reply:
> How are the HTTP headers from the original GET (index.html) request preserved on this request?

They are not. If original GET request headers are needed, then we would need to store them temporarily on Wave server and send them with the initial post Request to the Wave app. This should not be a problem.

The possible problem can arise during WS connection initialization since it's not possible to specify additional headers when initializing WS, only cookies are appended automatically. My guess is the proxy would reject such request as unauthorized. We can discuss on tomorrow's sync.